### PR TITLE
Testcontainers

### PR DESCRIPTION
**Pull request by SirNoob97**
* Basic test cases for the client repository 
* Couple of suggestions
* Update mongodb version in the docker-compose file

#2 #17

### CODE REVIEW

**Comment by SirNoob97** [comment]
This is my first time using mongo so I don't know if extending the TestContainer class of the config package is a good idea, but the it works.

File: cruise-ms-domain/src/test/java/com/example/cruisemsdomain/repository/ClientRepositoryTest.java [hunk 1]
```diff
@@ -22,13 +27,51 @@
 @DataMongoTest(excludeAutoConfiguration = EmbeddedMongoAutoConfiguration.class)
 public class ClientRepositoryTest extends TestContainer {
 
+	private static final Client CLIENT = createClient();
+
 	@Autowired
 	private ClientRepository clientRepository;
 
+	@AfterEach
+	public void deleteArterTest(){
```
Comment (kgotlelelo-cmd) [inline cruise-ms-domain/src/test/java/com/example/cruisemsdomain/repository/ClientRepositoryTest.java:36]: Typo

**Review by kgotlelelo-cmd — APPROVED** [review]
Wow.. I really appreciate your time and effort.
It's more than what i had in mind regarding testing great work.
Thanks i didn't know about test container library

**Comment by SirNoob97** [comment]
No problem. Now we must decide what kind of method the project needs for the client repository and then continue with the implementation of the tests.

**Comment by kgotlelelo-cmd** [comment]
I think we should use simple for student email
and not sure what more methods to add to repository.. but if possible return specific person's posts.
there is a bug when posting a post the post id is always zero i need help with.

please bring forth idea which might the project better
the project is mini twitter type of project with spring as domain layer that talks with data base and 
nodejs that call domain manipulate the data and sent to reactjs (no business logic will be implemented in reactjs) 
I am thinking of doing user authentication in nodejs I don't know if it going to be a good idea.

hopefully I understood your suggestion.

**Comment by SirNoob97** [comment]
Observing these resources #6, I notice that the embedded documents have no identification and that is probably the reason. And testing with Integer (not the primitive) and String, I see that the id does not persist in the database, so if you want a relationship between your entities, mongo is probably not a good idea, especially if you want to add more entities like comments.

I found this in the spring docs but it doesnt work as i expected [spring mongodb docs](https://docs.spring.io/spring-data/mongodb/docs/3.2.3/reference/html/#mapping-usage-references).

I know very well postgres if you decide to change the db engine, but if it is just for learning, we can move on with mongo.

**Comment by kgotlelelo-cmd** [comment]
I think you are right.
It is better to move to postgres(SQL) now rather than later..
as you mention it might block with future features. I  was my first time using it(it was for experimenting)

I have worked with sql and springboot before but it was a very small application, since you technically strong 
what suggestion or advise will you give me i.e regarding the structure of the database etc.

**Comment by kgotlelelo-cmd** [comment]
I think for  embedded documents with springboot you have to generate id for yourself

**Pull request merged**